### DISCUSSION TITLE
Using getserversideprops with Solito

### DISCUSSION THREAD

**Discussion by nikkwong** [nandorojo/solito#181] [Q&A] [answered]
I have an app that is using `getServerSideProps` to run logic including fetching user information, authentication status, and other tasks that are specific to the current session. It's obvious and straightforward on how to use this for the pages rendered with nextjs but I'm not sure how to do this for the native routes. Is this documented anywhere? Thanks!

**Comment by nikkwong** [comment]
*Bump* hoping for some help on this; thanks!

**Reply by kingfishx23** [reply]
Yeah... Anybody have figured this out?

**Comment by derek-primumco** [comment]
I'm pretty sure the concept of SSR doesn't exist in React Native yet (but Expo's latest SDK is moving toward static rendering).

You'd have to completely split off your rendering logic such that your "screens" are accepting the props. Then you'd have add logic to each page to pass the props (in Next.js via getServerSideProps and in React Native using some type of client-side data fetching like React Query).

In other words you're going to be able to share all code except the data fetching strategy.

<img width="608" alt="image" src="https://github.com/nandorojo/solito/assets/122907173/11b1f81f-4300-43ea-8dfb-b4ad869bf846">

**Comment by nandorojo** [comment, marked as answer]
use a shared screen across web and native with an optional data prop. on web, pass the result of gSSP to the screen. on native, fetch the data.
